File: v2/src/main/java/com/api/v2/people/dtos/Address.java
package com.api.v2.people.dtos;

import com.api.v2.common.States;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;

public record Address(
        @NotBlank
        String street,
        @NotBlank
        String city,
        @NotNull
        States state,
        @NotBlank
        @Size(min = 5, max = 5)
        String zipcode
) {
}

File: v2/src/main/kotlin/com/api/v2/people/dtos/Address.kt
package com.api.v2.people.dtos

import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.Size

data class Address(
    @NotBlank
    val street: String,
    @NotBlank
    val city: String,
    @NotBlank
    @Size(min = 5, max = 5)
    val zipcode: String
)